components/brave_wallet/browser/sns_resolver_task.cc

```diff
@@ -0,0 +1,665 @@
+  filters.back().GetDict().SetByDottedPath("memcmp.offset", 64);
+  filters.back().GetDict().SetByDottedPath("memcmp.bytes",
+                                           "2");  // base58 of 0x01
+  filters.Append(base::Value::Dict());
+  // https://github.com/solana-labs/solana-program-library/blob/f97a3dc7cf0e6b8e346d473a8c9d02de7b213cfd/token/program/src/state.rs#L129
+  filters.back().GetDict().Set("dataSize", 165);
+  configuration.Set("filters", std::move(filters));
+
+  params.Append(std::move(configuration));
+
+  base::Value::Dict dictionary =
+      GetJsonRpcDictionary("getProgramAccounts", std::move(params));
+  return GetJSON(dictionary);
+}
+
+// https://docs.solana.com/developing/clients/jsonrpc-api#example-34
+// Parsing result of getProgramAccounts call. Exepected to find 1 token account
+// for mint. If parsing fails first element of pair is `false`.
+std::pair<bool, absl::optional<SolanaAddress>>
+GetTokenOwnerFromGetProgramAccountsResult(const std::string& json) {
+  auto result = ParseResultList(json);
+  if (!result)
+    return {false, absl::nullopt};
+
+  if (result->size() != 1)
+    return {true, absl::nullopt};
+
+  auto* item = result->front().GetIfDict();
+  if (!item)
+    return {false, absl::nullopt};
+  auto* account = item->FindDict("account");
+  if (!account)
+    return {false, absl::nullopt};
+
+  absl::optional<SolanaAccountInfo> account_info;
+  if (!solana::ParseGetAccountInfo(*account, &account_info)) {
+    return {false, absl::nullopt};
+  }
+
+  DCHECK(account_info);
+
+  auto account_data = FromBase64<SplAccountData>(account_info->data);
+  if (!account_data)
+    return {false, absl::nullopt};
+
+  return {true, account_data->owner};
+}
+
+}  // namespace
+
+SnsNamehash GetHashedName(const std::string& name) {
+  // https://github.com/Bonfida/solana-program-library/blob/6e3be3eedad3a7f4a83c1b7cd5f17f89231e0bca/name-service/js/src/constants.ts#L13
+  constexpr char kHashPrefix[] = "SPL Name Service";
+  const std::string input = kHashPrefix + name;
+  return crypto::SHA256Hash(base::as_bytes(base::make_span(input)));
+}
+
+absl::optional<SolanaAddress> GetNameAccountKey(
+    const SnsNamehash& hashed_name,
+    const absl::optional<SolanaAddress>& parent) {
+  if (!parent)
+    return absl::nullopt;
+
+  std::vector<std::vector<uint8_t>> seeds;
+  seeds.emplace_back(hashed_name.begin(), hashed_name.end());
+  seeds.emplace_back(32, 0);  // we don't use nameClass here.
+  seeds.emplace_back(parent->bytes());
+
+  // https://github.com/Bonfida/solana-program-library/blob/6e3be3eedad3a7f4a83c1b7cd5f17f89231e0bca/name-service/js/src/constants.ts#L7
+  constexpr char kNameProgramId[] =
+      "namesLPneVptA9Z5rqUDD9tMTWEJwofgaYwp8cawRkX";
+
+  auto address =
+      SolanaKeyring::FindProgramDerivedAddress(seeds, kNameProgramId);
+  if (!address)
+    return absl::nullopt;
+
+  return SolanaAddress::FromBase58(*address);
+}
+
+absl::optional<SolanaAddress> GetMintAddress(
```

Comment:
Link to https://github.com/Bonfida/name-tokenizer#mint

Reply:
https://github.com/brave/brave-core/pull/15617/commits/1e34bd9363c2e1aa4593c00f3409b8c119ccab29